### PR TITLE
20200105_dungdq: commit login

### PR DESCRIPTION
**Pull request by dqdung1230**

### CODE REVIEW

File: mockup-service-report/src/main/java/com/itsol/mockup/entity/EmployeeEntity.java [hunk 1]
```diff
@@ -14,8 +15,9 @@
 @NoArgsConstructor
 @ToString
 @FieldDefaults(level = AccessLevel.PRIVATE)
+public class EmployeeEntity implements Serializable {
+
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/entity/EmployeeEntity.java:19]: remove blank line

File: mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java [hunk 2]
```diff
@@ -1,18 +1,21 @@
 package com.itsol.mockup.web.dto.response;
 
 import com.itsol.mockup.utils.Constants;
+import lombok.AccessLevel;
+import lombok.Data;
 import lombok.Getter;
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java]: remove unused import

File: mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
 import lombok.Setter;
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java]: remove unused import

File: mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
 
 /**
  * @author anhvd_itsol
  */
 
-@Getter
-@Setter
+@Data
+@FieldDefaults(level = AccessLevel.PROTECTED)
 public class BaseResponseDTO {
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java:16]: remove toString() method add @ToString, add @AllArgConstructor & @NoArgConstructor

File: mockup-service-report/src/main/java/com/itsol/mockup/web/rest/users/EmployeeController.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+/**
+ * @author anhvd_itsol
+ */
+
+@RestController
+@RequestMapping("/api")
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/web/rest/users/EmployeeController.java]: it should be change to "/employee"

File: mockup-service-report/src/main/resources/application.properties [hunk 6]
```diff
@@ -1,15 +1,12 @@
...
-
-spring.datasource.hibernate.dialect=org.hibernate.dialect
-spring.jpa.hibernate.ddl-auto = update
-spring.main.allow-bean-definition-overriding=true
+spring.jpa.database-platform=org.hibernate.dialect.OracleDialect
+#spring.jpa.hibernate.ddl-auto = update
+#spring.main.allow-bean-definition-overriding=true
 spring.jpa.database=default
 spring.jpa.show-sql=true
 server.port=8008
```
Comment (truongbb) [inline mockup-service-report/src/main/resources/application.properties:11]: config api URL here: data.rest.base-path: '<value>'

File: mockup-service-report/src/main/java/com/itsol/mockup/repository/impl/EmployeeRepositoryImpl.java [hunk 7]
```diff
@@ -28,11 +28,12 @@
  */
 
 @Repository
-public class UsersRepositoryImpl implements UsersRepositoryCustom {
-    private static final Logger logger = LogManager.getLogger(UsersRepositoryImpl.class);
+public class EmployeeRepositoryImpl implements EmployeeRepositoryCustom {
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/repository/impl/EmployeeRepositoryImpl.java:31]: line 55->59, 64->68 should be written under a method

File: mockup-service-report/src/main/java/com/itsol/mockup/repository/EmployeeRepository.java [hunk 8]
```diff
@@ -11,9 +11,10 @@
...
 
 @Repository
-public interface UsersRepository extends JpaRepository<EmployeeEntity, Long> {
-    EmployeeEntity getUsersById (Long id);
-    Page<EmployeeEntity> getAllByUserNameAndFullName(String userName, String fullName, Pageable pageable);
+public interface EmployeeRepository extends JpaRepository<EmployeeEntity, Long> {
+    EmployeeEntity getUsersById(Long id);
 
-    EmployeeEntity getUsersByUserName(String username); //function for generate token
+    Page<EmployeeEntity> getAllByUsernameAndFullName(String username, String fullName, Pageable pageable);
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/repository/EmployeeRepository.java]: remove unused method

File: mockup-service-report/src/main/java/com/itsol/mockup/services/impl/UsersServiceImpl.java [hunk 9]
```diff
@@ -109,18 +114,18 @@
...
-            EmployeeEntity user = usersRepository.getUsersById(usersDTO.getId());
+            EmployeeEntity user = employeeRepository.getUsersById(employeeDTO.getId());
             if (user.getId() != null) {
-                user.setUsername(usersDTO.getUserName());
-                user.setFullName(usersDTO.getFullName());
-                user.setPassword(usersDTO.getPassWord());
+                user.setUsername(employeeDTO.getUsername());
+                user.setFullName(employeeDTO.getFullName());
+                user.setPassword(employeeDTO.getPassword());
 //                user.setRoleEntities(usersDTO.getRole());
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/services/impl/UsersServiceImpl.java]: check roles and set here

File: mockup-service-report/src/main/java/com/itsol/mockup/services/impl/UsersServiceImpl.java [hunk 10]
```diff
@@ -1,29 +1,34 @@
...
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.PageRequest;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
+import java.util.HashSet;
```
Comment (truongbb) [inline mockup-service-report/src/main/java/com/itsol/mockup/services/impl/UsersServiceImpl.java]: remove unused import in this class

File: mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java (hunk 3 above)
Comment (dqdung1230) [reply mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java]: done

File: mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java (hunk 4 above)
Comment (dqdung1230) [reply mockup-service-report/src/main/java/com/itsol/mockup/web/dto/response/BaseResponseDTO.java:16]: done

File: mockup-service-report/src/main/java/com/itsol/mockup/web/rest/users/EmployeeController.java (hunk 5 above)
Comment (dqdung1230) [reply mockup-service-report/src/main/java/com/itsol/mockup/web/rest/users/EmployeeController.java]: done

File: mockup-service-report/src/main/resources/application.properties (hunk 6 above)
Comment (dqdung1230) [reply mockup-service-report/src/main/resources/application.properties:11]: done

File: mockup-service-report/src/main/java/com/itsol/mockup/services/impl/UsersServiceImpl.java (hunk 10 above)
Comment (dqdung1230) [reply mockup-service-report/src/main/java/com/itsol/mockup/services/impl/UsersServiceImpl.java]: ok

File: mockup-service-report/src/main/java/com/itsol/mockup/repository/EmployeeRepository.java (hunk 8 above)
Comment (dqdung1230) [reply mockup-service-report/src/main/java/com/itsol/mockup/repository/EmployeeRepository.java]: ok

File: mockup-service-report/src/main/java/com/itsol/mockup/repository/impl/EmployeeRepositoryImpl.java (hunk 7 above)
Comment (dqdung1230) [reply mockup-service-report/src/main/java/com/itsol/mockup/repository/impl/EmployeeRepositoryImpl.java:31]: ok

File: mockup-service-report/src/main/java/com/itsol/mockup/services/impl/UsersServiceImpl.java (hunk 9 above)
Comment (truongbb) [reply mockup-service-report/src/main/java/com/itsol/mockup/services/impl/UsersServiceImpl.java]: maybe later

**Pull request merged**